pkg/simulation/sim-iteration.go

```diff
@@ -0,0 +1,57 @@
+package simulation
+
+import (
+	"sync"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/agent"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/world"
+)
+
+func (sE *SimEnv) SetWorld(w world.World) {
+	if sE.world == nil {
+		sE.world = w
+	}
+}
+
+func (sE *SimEnv) World() world.World {
+	return sE.world
+}
+
+func (sE *SimEnv) simulationLoop(t *infra.Tower) {
+	for sE.dayInfo.CurrTick <= sE.dayInfo.TotalTicks {
+		sE.replaceAgents(t)
+		sE.TowerTick()
+		sE.AgentsRun()
+		if sE.reportFunc != nil {
+			sE.reportFunc(sE)
+		}
+		sE.dayInfo.CurrTick++
+	}
+}
+
+func (sE *SimEnv) SetReportFunc(fn func(*SimEnv)) {
+	sE.reportFunc = fn
```

Comment:
Is this supposed to be called anywhere?

Reply:
Loool I've got not idea - I've baso copied and pasted it from ABM iterate lol.

Reply:
In that case, I think delete it - no point in having it unless we know we need it.

We can always add it back later.